### DISCUSSION TITLE
Can modify `res` and `req` at initialization or with like app.__proto__.res?

### DISCUSSION THREAD

**Discussion by dalisoft** [uNetworking/uWebSockets.js#152] [Q&A]
Hi @alexhultman 

Can modify `res` and `req` at initialization or with like `app.__proto__.res`?

Like this possible? I couldn't find any related issue

```js
uWebSockets.App.__proto__.HttpResponse.setHeader = function (key, value) {
this.writeHeader(key, value);
}
```

Doing this when request (modify in handler possible, but very bad performance) drops performance by 2x-3x which not good

Allowing users modify `HttpResponse` and `HttpRequest` allows doing more complex stuff with small loss of performance instead of 2-3x performance :)

UPD 1: I improved performance by handling when needed, now my requests almost 30-40% slower than uWebSockets.js, but it's still slow (but fastest Node.js server framework after uWebSockets.js with Connect/Express-like API with schema support)

UPD 2: You can see my lib at [here](https://github.com/dalisoft/nanoexpress), i already linked this library as i'm made on top of uWebSockets.js

**Comment by ghost** [comment]
What's wrong with doing this?

```js
class MyHttpResponse {
    constructor(httpResponse) {
        this.httpResponse = httpResponse;
    }

    setHeader(key, value) {
        httpResponse.writeHeader(key, value);
    }
}

new MyHttpResponse(httpResponse);
```

**Comment by dalisoft** [comment]
Thanks for response.
Nothing wrong, i am did like this, even performant than this. But this or my way drops performance by 20-30%.
